### ISSUE TITLE
differing info in "supported by"

### ISSUE THREAD

**Issue by mtholder** [OpenTreeOfLife/treemachine#180]
When I load:
  https://devtree.opentreeoflife.org/opentree/opentree3.0@3846678
and look at the "supported by" info for the "Edge to parent Phaseoleae" info I see 6 trees and the taxonomy.  

When I load https://devtree.opentreeoflife.org/opentree/opentree3.0@369901/Phaseoleae
and ask for the same edge info (edge leading to Glycine + Phaseolus+ Vigna) I only see 3 trees supporting it.

*note* this is one of the nodes that otcetera flags as unsupported, fwiw.

**Comment by mtholder** [comment]
Another example is https://devtree.opentreeoflife.org/opentree/opentree3.0@3870247 
where  pg_2838 drops from the supporting list if you move back a node.

**Comment by jimallman** [comment]
It seems to be a bug in the arguson. For some nodes, we seem to get additional support if it's the target (primary) node versus elsewhere in the tree. In calls to `getSyntheticTree`, the target node is specified as `subtreeNodeID`. 

I tested this with node _opentree3.0 @ 3846678_ on **devtree**. Here's the curl call to fetch its subtree in arguson:
```bash
curl 'https://devapi.opentreeoflife.org/cached/treemachine/v1/getSyntheticTree' -H 'Origin: https://devtree.opentreeoflife.org' -H 'Accept-Encoding: gzip, deflate' -H 'Accept-Language: en-US,en;q=0.8' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.152 Safari/537.36' -H 'Content-Type: application/json' -H 'Accept: text/plain, */*; q=0.01' -H 'Referer: https://devtree.opentreeoflife.org/opentree/argus/opentree3.0@3846678' -H 'Connection: keep-alive' --data-binary '{"treeID":"opentree3.0","format":"arguson","maxDepth":"3","subtreeNodeID":"3846678"}' --compressed
```
For the target node, this returns seven studies plus taxonomy:
```json
      "supportedBy" : [ 
            "pg_1087_2115_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "pg_2690_6243_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "pg_2712_6296_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "pg_2820_6566_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "pg_2057_4240_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "pg_2539_6294_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "taxonomy", 
            "pg_1087_2114_a2c48df995ddc9fd208986c3d4225112550c8452" 
      ],
```

Now we fetch the arguson for its parent node:
```bash
curl 'https://devapi.opentreeoflife.org/cached/treemachine/v1/getSyntheticTree' -H 'Origin: https://devtree.opentreeoflife.org' -H 'Accept-Encoding: gzip, deflate' -H 'Accept-Language: en-US,en;q=0.8' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.152 Safari/537.36' -H 'Content-Type: application/json' -H 'Accept: text/plain, */*; q=0.01' -H 'Referer: https://devtree.opentreeoflife.org/opentree/argus/opentree3.0@369901/Phaseoleae' -H 'Connection: keep-alive' --data-binary '{"treeID":"opentree3.0","format":"arguson","maxDepth":"3","subtreeNodeID":"369901"}' --compressed
```
The resulting arguson shows a smaller list of supporting studies (a **subset** of the above) for node _3846678_:
```json
        "supportedBy" : [ 
            "pg_2712_6296_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "pg_2820_6566_a2c48df995ddc9fd208986c3d4225112550c8452", 
            "pg_2539_6294_a2c48df995ddc9fd208986c3d4225112550c8452" 
        ],
```

I'm not sure which set of supporting studies is accurate. I'm bouncing this to @josephwb for a closer look, since he recently added the improved support. Looks like [something funny is going on in `GraphExplorer.reconstructSyntheticTreeHelper`](https://github.com/OpenTreeOfLife/treemachine/blob/2d0195f8d92bbb4e2c6a8f944b96ed2c45d2e050/src/main/java/opentree/GraphExplorer.java#L3570)

**Comment by josephwb** [comment]
Thanks @jimallman

**Comment by josephwb** [comment]
FWIW the ```node_status``` services gives the following:
```
curl X POST http://devapi.opentreeoflife.org/v2/graph/node_info -H "content-type:application/json" -d '{"node_id":3846678}'
```
```json
{
  "tree_id" : "opentree3.0",
  "num_synth_tips" : 387,
  "name" : "",
  "rank" : "",
  "ott_id" : "null",
  "num_tips" : 297,
  "tree_sources" : [ {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6296",
    "study_id" : "pg_2712"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6566",
    "study_id" : "pg_2820"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6294",
    "study_id" : "pg_2539"
  } ],
  "tax_source" : "",
  "synth_sources" : [ {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "2115",
    "study_id" : "pg_1087"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6243",
    "study_id" : "pg_2690"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6296",
    "study_id" : "pg_2712"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6566",
    "study_id" : "pg_2820"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "4240",
    "study_id" : "pg_2057"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6294",
    "study_id" : "pg_2539"
  }, {
    "git_sha" : "",
    "tree_id" : "",
    "study_id" : "taxonomy"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "2114",
    "study_id" : "pg_1087"
  } ],
  "node_id" : 3846678,
  "in_synth_tree" : true
}
```
So, "tree_sources" and "synth_sources" differ. Will see what is going on and fix things.

**Comment by josephwb** [comment]
I think it is "tree_sources" that is wrong. Here is an example from birds (what else?):
```
curl -X POST http://devapi.opentreeoflife.org/v2/graph/node_info -H "content-type:application/json -d '{"ott_id":81461}'
```
```json
{
  "tree_id" : "opentree3.0",
  "num_synth_tips" : 14088,
  "name" : "Aves",
  "rank" : "class",
  "ott_id" : 81461,
  "num_tips" : 14088,
  "tree_sources" : [ ],
  "tax_source" : "ncbi:8782,gbif:212,irmng:1142",
  "synth_sources" : [ {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "522",
    "study_id" : "pg_420"
  }, {
    "git_sha" : "a2c48df995ddc9fd208986c3d4225112550c8452",
    "tree_id" : "6757",
    "study_id" : "pg_2926"
  }, {
    "git_sha" : "",
    "tree_id" : "",
    "study_id" : "taxonomy"
  } ],
  "node_id" : 788774,
  "in_synth_tree" : true
}
```

**Comment by jimallman** [comment]
Are `tree_sources` and `synth_sources` documented? The [API wiki entry for `node_info`](https://github.com/OpenTreeOfLife/opentree/wiki/Open-Tree-of-Life-APIs#node_info) doesn't cover this level of detail, and the [source for `getSynthesisSupportingSources` and `getSupportingTreeSources` doesn't really spell it out either](https://github.com/OpenTreeOfLife/treemachine/blob/2d0195f8d92bbb4e2c6a8f944b96ed2c45d2e050/src/main/java/opentree/plugins/graph.java#L112), at least to my eye.

**Comment by josephwb** [comment]
Thanks for pointing out the lag in the doc @jimallman. 

```tree_sources``` is _supposed_ to be for nodes _not_ in the synthetic tree (i.e. they might have tree support, but are overruled because of conflict with other trees; this is a step towards the goal of explaining why a particular node does not appear in the synthetic tree), while ```synth_sources``` is _supposed_ to be for nodes that _are_ in the synthetic tree. If we are talking about a node that _is_ in the synthetic tree, the lists should be the same.

I am not sure what is going on with ```tree_sources```; it is clearly missing trees that do in fact support them. I feel that ```synth_sources``` is reliable. I can update this particular service to only report ```synth_sources```, since it will only ever deal with nodes that are in the path of the synthetic tree. That said, it is disturbing that ```tree_sources``` are missed...

**Comment by josephwb** [comment]
Found the problem. Will deploy a fix this afternoon to dev.

**Comment by josephwb** [comment]
Should be fixed with f5f9b94140bfe28e2bbaffdd5c81ffacad4d985f. Please test @jimallman.
